src/content/docs/it/guides/upgrade-to/v2.mdx

```diff
@@ -0,0 +1,423 @@
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+
+export default defineConfig({
+  markdown: {
+    remarkPlugins: [remarkPlugin1],
+    gfm: true,
+  },
+  integrations: [
+    mdx({
+      remarkPlugins: [remarkPlugin2],
+      gfm: false,
+    })
+  ]
+});
+```
+
+##### Cosa devo fare?
+
+Rivedi le configurazioni di Markdown e MDX e confronta la tua configurazione con le nuove opzioni disponibili.
+
+
+### Cambiato: Accesso dei plugn all'avantesto
+
+Nella v1.x, i plugin remark e rehype non avevano accesso all'avantesto dell'utente. Astro ha unito l'avantesto del plugin con l'avantesto del tuo file, senza dover passare l'avantesto del file ai tuoi plugin.
+
+Astro v2.0 consente ai plugin remark e rehype l'accesso all'avantesto degli utenti tramite l'iniezione di avantesto. Ciò consente agli autori di plug-in di modificare l'avantesto esistente di un utente o di calcolare nuove proprietà basate su altre proprietà.
+
+
+#### Cosa devo fare?
+
+Controlla qualsiasi plugin remark e rehype che hai scritto per vedere se il loro comportamento sia cambiato. Nota che `data.astro.frontmatter` è adesso l'_intero_ avantesto del documento Markdown o MDX, piuttosto che un oggetto vuoto.
+
+
+### Cambiato: Configurazione di RSS
+
+Nella v1.x, il pacchetto RSS di Astro permetteva di usare `items: import.meta.glob(...)` per generare un elenco di elementi del feed RSS. Questo utilizzo è ora obsoleto e alla fine sarà rimosso.
+
+Astro v2.0 introduce un `pagesGlobToRssItems()` wrapper alla proprietà `items`.
+
+
+#### Cosa devo fare?
+
+Importa, poi avvolgi la funzione esistente contenente `import.meta.glob()` con il helper `pagesGlobToRssItems()`.
+
+```js ins={3, 8, 10}
+// src/pages/rss.xml.js
+import rss, {
+  pagesGlobToRssItems
+} from '@astrojs/rss';
+
+export async function get(context) {
+  return rss({
+    items: await pagesGlobToRssItems(
+      import.meta.glob('./blog/*.{md,mdx}'),
+    ),
+  });
+}
+```
+
+### Cambiato: Supporto di Svelte per l'IDE
+
+Astro v2.0 necessita del file `svelte.config.js` nel tuo progetto se stai usando [l'integrazione `@astrojs/svelte`](/it/guides/integrations-guide/svelte/). Ciò è necessario per fornire il completamento automatico dell'IDE.
+
+#### Cosa devo fare?
+
+Aggiungi il file `svelte.config.js` alla radice del tuo progetto:
+
+```js
+// svelte.config.js
+import { vitePreprocess } from '@astrojs/svelte';
+
+export default {
+  preprocess: vitePreprocess(),
+};
+```
+
+Per i nuovi utenti, questo file sarà aggiunto automaticamente con l'esecuzione di `astro add svelte`.
+
+### Rimosso: `legacy.astroFlavoredMarkdown`
+
+Nella v1.0, Astro ha spostato il vecchio Astro-Flavored Markdown (conosciuto anche come Components in Markdown) ad una vecchia funzionalità.
```

Comment:
**nitpick:** I would rephrase this sentence to be more fluent.

Già nella v1.0, Astro aveva dichiarato come deprecato il vecchio Astro-Flavored Markdown (conosciuto anche come Components in Markdown).